### ISSUE TITLE
Session#Add does not return any information about what was inserted

### ISSUE THREAD

**Issue by proc** [aacanakin/qb#59]
Is there a way to insert a single row, returning that row that was inserted? It would be very useful to get an ID back after inserting a row (using Returning). It doesn't seem possible to do this within the Session. Is using the Builder the only way of doing this?

**Comment by aacanakin** [comment]
There is an internal builder object in session. Maybe, that can be exposed using a `Builder()` function as in the following;
```go
session := qb.New("postgres", "blah blah")
id := session.Builder().Insert("users").Values(map[string]interface{}{"id": 5}).Returning("id").Exec()
```
This option looks good but it's not using session's struct mapping techniques. It's the easiest implementation.

The second option that comes to mind is to add optional returning parameters as in the following;
```go
db.Add(user, "id", "email")
```
This looks ok however, since the returning clause is postgres specific, the `Add` function's optional parameters would be completely useless if you're using mysql or sqlite.

The third option would be manually adding primary key fields after add is executed if the driver is postgres. In that implementation, the returning clause would be automatically generated for only primary key fields.

Which one do you prefer? Can you suggest another solution?

**Comment by proc** [comment]
Hey @aacanakin thanks for the response. Personally I prefer something similar to the second option, but not exactly. When you call db.Add you pass in a struct, which I think could be updated with the returning values after you commit. You could return * and use those values. Or maybe create another instance of the struct just to capture the return values, if we're afraid of modifying the original struct that's passed in. 

example usage:

user := &User{ID: 0, Name: "John"}
db.Add(user)

user.ID now equals whatever the ID was after insert.

**Comment by aacanakin** [comment]
The `Add()` function only creates a transaction and adds insert statements to the transaction in progress.  You need to call `Commit()` after `Add()` to persist it in db. Therefore, it's hard to fill the struct in `Add()`. I'll come up with a solution to that. It'll be in 0.2 beta release.

**Comment by aacanakin** [comment]
Currently, in the master branch, there exists the new expression api implementation which have Returning clause in inserts, upserts, deletes & updates. Here's the example;

```go
package main

import (
	"fmt"
	"github.com/aacanakin/qb"
)

func main() {

	type User struct {
		ID   int64  `qb:"type:bigserial; constraints:primary_key"`
		Name string `qb:"constraints:not_null"`
	}

	db, err := qb.New("postgres", "user=postgres dbname=qb_test sslmode=disable")
	checkErr(err)

	db.Dialect().SetEscaping(true)

	db.AddTable(User{})
	err = db.CreateAll()
	defer db.DropAll()
	checkErr(err)

	ins := db.T("user").
		Insert().
		Values(map[string]interface{}{
			"name": "John",
		}).
		Returning(db.T("user").C("id"), db.T("user").C("name")).
		Build(db.Dialect())

	var user User
	err = db.Engine().Get(ins, &user)
	checkErr(err)

	fmt.Printf("<User id=%d name=%s/>\n", user.ID, user.Name)
}

func checkErr(err error) {
	if err != nil {
		panic(err)
	}
}
```
In the new expression api, there exists `T(name string)` & `C(name string)` functions. `T()` returns the table object registered to metadata & `C()` returns the column object of that table.

One disadvantage is; This statement would not run in a transaction.
Currently, I couldn't come up with a solution that works in session api. Would that be ok for you @proc ?

**Comment by aacanakin** [comment]
session.Add() would not be available in the 0.3 release. Only expression api & engine query executors will be available.
